### ISSUE TITLE
Simultaneous table index creation/deletion limited to one-per-table

### ISSUE THREAD

**Issue by alexdilley** [architect/architect#268]
**Describe the bug**
Issuing `npx create` where more than one index on a table is to be created will fail: AWS appears to limit DDB index creation to "only 1 online index [can be created or deleted] simultaneously per table".

**To Reproduce**
Fresh `npx create` on a project that with an equivalent `.arc` manifest:
```
@app
testapp

@tables
data
  PK *String
  SK **String

@indexes
data
  GSI1PK *String
  GSI1SK **String

data
  GSI2PK *String
  GSI2SK **String
```

Error:

```
{ LimitExceededException: Subscriber limit exceeded: Only 1 online index can be created or deleted simultaneously per table
    at Request.extractError (/path/to/arc/project/node_modules/aws-sdk/lib/protocol/json.js:51:27)
    at Request.callListeners (/path/to/arc/project/node_modules/aws-sdk/lib/sequential_executor.js:106:20)
    at Request.emit (/path/to/arc/project/node_modules/aws-sdk/lib/sequential_executor.js:78:10)
    at Request.emit (/path/to/arc/project/node_modules/aws-sdk/lib/request.js:683:14)
    at Request.transition (/path/to/arc/project/node_modules/aws-sdk/lib/request.js:22:10)
    at AcceptorStateMachine.runTo (/path/to/arc/project/node_modules/aws-sdk/lib/state_machine.js:14:12)
    at /path/to/arc/project/node_modules/aws-sdk/lib/state_machine.js:26:10
    at Request.<anonymous> (/path/to/arc/project/node_modules/aws-sdk/lib/request.js:38:9)
    at Request.<anonymous> (/path/to/arc/project/node_modules/aws-sdk/lib/request.js:685:12)
    at Request.callListeners (/path/to/arc/project/node_modules/aws-sdk/lib/sequential_executor.js:116:18)
  message:
   'Subscriber limit exceeded: Only 1 online index can be created or deleted simultaneously per table',
  code: 'LimitExceededException',
  time: 2019-01-17T16:18:39.279Z,
  requestId: 'GFBQ9OSFGQIVF6AI0756P78O3JVV4KQNSO5AEMVJF66Q9ASUAAJG',
  statusCode: 400,
  retryable: false,
  retryDelay: 44.153841958458365 }
```

**Expected behavior**
Index creation _per table_ to be queued. The workaround currently is to simply re-execute the `create` command.

**Comment by brianleroux** [comment]
bizarre I haven't seen that… wonder if they changed the behavior. easy fix / we can queue those operations!

**Comment by alexdilley** [comment]
The difficulty with queuing will be that indexes take _a long time_ to create. (A further index cannot be created even if when another is in the the state of `Creating`.)

**Comment by ryanblock** [comment]
Come to think of it, we experienced aspects of this issue internally recently when we refactored Begin onto Architect 4.x.

You're def right, indexes take a long time to create, and depending on the table, it's unpredictable. However, we could queue the operation by periodically polling the status of the last index creation before attempting the next. Thoughts?

**Comment by brianleroux** [comment]
thx for reporting this; fix published in `@architect/architect@5.8.3`
